### PR TITLE
Added cgi

### PR DESCRIPTION
**Pull request by Gerhard-Vienna**
Hi Chris!

Here is the added code.

Beside the added functionality I had to make some minor changes to make it compile on my system (Debian 11):
in file ./CmakeList.txt: I changed the required cmake version to 3.18, 
in file ./src/CmakeList.txt: I had to comment out the line COMMAND_ERROR_IS_FATAL ANY

To make better use of the pico-probe programmer I changed "pico_enable_stdio_uart" to "true" in ./src/CmakeList.txt
I also took the liberty to "improve" your web pages...

Change that  back, if you think it's not a good idea ;-)

Gerhard

### CODE REVIEW

File: src/cgi.h [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  *
+  * Copyright (c) 2017 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
```
Comment (krzmaz) [inline src/cgi.h]: I'm not really in favor of adding the file with a specific author name and a different license.  What if me or someone else wants to modify/extend it later? It makes maintenance far more complex. There is a common LICENSE file in the root of this repository (it's MIT though).

If you're worried about the STMicroelectronics copyright, why not derive the example implementation from the lwIP example like the other examples in this repository?
https://github.com/lwip-tcpip/lwip/blob/master/contrib/examples/httpd/cgi_example

File: src/main.cpp [hunk 2]
```diff
@@ -6,11 +6,13 @@
 #include "pico/cyw43_arch.h"
 
 #include "lwipopts.h"
+#include "cgi.h"
 #include "ssi.h"
 
 void run_server() {
     httpd_init();
     ssi_init();
+	cgi_init();
```
Comment (krzmaz) [inline src/main.cpp]: nit-pick: please fix the indent here
```suggestion
    cgi_init();
```

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -14,7 +14,7 @@ execute_process(COMMAND
         WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
         ECHO_OUTPUT_VARIABLE
         ECHO_ERROR_VARIABLE
-        COMMAND_ERROR_IS_FATAL ANY
+#        COMMAND_ERROR_IS_FATAL ANY
```
Comment (krzmaz) [inline src/CMakeLists.txt]: Let's just remove this line then, this will change anyway once I get back to #4 
```suggestion

```

File: .gitignore [hunk 4]
```diff
@@ -6,3 +6,7 @@ cmake/pico_sdk_import.cmake
 # clangd stuff
 .cache
 compile_commands.json
+
+#kate backups
+*~
+
```
Comment (krzmaz) [inline .gitignore:12]: I think that was added by mistake

**Review by krzmaz — COMMENTED** [review]
Other than two small suggestions, it looks good to me.
(You can apply those suggestions using Github)

File: .gitignore (hunk 4 above)
Comment (krzmaz) [reply .gitignore:12]: Nevermind, I didn't realize that Kate was an editor 😅

**Comment by Gerhard-Vienna** [comment]
I believe it is done now. 
If you want anything changed, let me know.

**Comment by krzmaz** [comment]
Looks great, thanks for this! 😄

**Pull request merged**